File: rulepacks/2026.08.2/testdata/house-rules.cs
// Fixture de semgrep --test — no es código del producto.
using System;

namespace Fixtures
{
    public class ReglasCasaFixture
    {
        public void Procesar()
        {
            // ruleid: csharp-empty-catch
            try { EjecutarTarea(); } catch (Exception e) { }
        }

        public void ProcesarBien()
        {
            // ok: csharp-empty-catch
            try { EjecutarTarea(); } catch (Exception e) { Console.Error.WriteLine(e); }
        }

        public string CadenaConexion()
        {
            // ruleid: hardcoded-connstring
            return "Server=x;Database=demo;Password=changeme";
        }

        public string CadenaConexionBien()
        {
            // ok: hardcoded-connstring
            return Environment.GetEnvironmentVariable("CADENA_CONEXION");
        }

        private void EjecutarTarea()
        {
            Console.WriteLine("tarea");
        }
    }
}

File: rulepacks/2026.08.3/testdata/house-rules.java
// Fixture de semgrep --test — no es código del producto.
class ReglasCasaFixture {
    void procesar() {
        // ruleid: java-empty-catch
        try {
            ejecutarTarea();
        } catch (Exception e) {
        }
    }

    void procesarBien() {
        // ok: java-empty-catch
        try {
            ejecutarTarea();
        } catch (Exception e) {
            System.err.println("fallo la tarea: " + e.getMessage());
        }
    }

    void ejecutarTarea() {
        System.out.println("tarea");
    }
}
